jetty-util/src/main/java/org/eclipse/jetty/util/DateCache.java

```diff
@@ -22,57 +22,60 @@
 import java.util.TimeZone;
 
 /**
- * Date Format Cache.
- * Computes String representations of Dates and caches
- * the results so that subsequent requests within the same second
- * will be fast.
- *
- * Only format strings that contain either "ss".  Sub second formatting is
- * not handled.
- *
- * The timezone of the date may be included as an ID with the "zzz"
- * format string or as an offset with the "ZZZ" format string.
- *
+ * Computes String representations of Dates then caches the results so
+ * that subsequent requests within the same second will be fast.
+ * <p>
+ * When formatting with the cache, sub second formatting will not be precise
+ * and may use a cached result.
+ * <p>
```

Comment:
I think there should be a mention that, IIUC, this class is meant to cache only the most recent dates, since its main user is `CustomRequestLog`, where requests times are almost always monotonically growing.

In 12 perhaps we can get rid of this class and just move what's necessary in `CustomRequestLog`, as the other users of `DateCache` seem to not require the caching (and possibly pollute the cache).

Reply:
There is similar code in `Server.getDateField()`.  Perhaps this can be generalised to allow caching of anything that can be generated from the current time?